python/ray/serve/tests/test_api.py

```diff
@@ -409,6 +409,30 @@ def __call__(self, *args):
     assert ray.get(ingress_handle.remote()) == "got f"
 
 
+def test_run_delete_old_deployments(serve_instance):
+    """Check that serve.run() can remove all old deployments"""
+
+    @serve.deployment(name="f", route_prefix="/f")
+    def f():
+        return "got f"
+
+    @serve.deployment(name="g", route_prefix="/g")
+    def g():
+        return "got g"
+    
+    app = Application([f])
```

Comment:
can you use the `.bind()` API instead of `Application`? sorry I know this is probably confusing, there is some tech debt cleanup in flight...

e.g., `serve.run(f.bind())`